src/main/kotlin/sg/com/quantai/etl/services/TimescaleDBTableInitializer.kt

```diff
@@ -34,37 +34,45 @@ class TimescaleDBTableInitializer(val jdbcTemplate: JdbcTemplate) {
     fun runDatabaseInitializer(): CommandLineRunner {
         return CommandLineRunner {
 
-            // Start of table initialization
             logger.info("Initializing all required tables for TimescaleDB...")
 
-            // Initialize `raw_crypto_data` table
             logger.info("Initializing 'raw_crypto_data' table...")
             jdbcTemplate.execute("""
                 CREATE TABLE IF NOT EXISTS raw_crypto_data (
                     id SERIAL PRIMARY KEY,
                     symbol VARCHAR(10) NOT NULL,
-                    source VARCHAR(50) NOT NULL,
-                    price DECIMAL NOT NULL,
-                    timestamp TIMESTAMPTZ NOT NULL
+                    currency VARCHAR(10) NOT NULL,
+                    open DECIMAL NOT NULL,
+                    high DECIMAL NOT NULL,
+                    low DECIMAL NOT NULL,
+                    close DECIMAL NOT NULL,
+                    volume_from DECIMAL NOT NULL,
+                    volume_to DECIMAL NOT NULL,
+                    timestamp TIMESTAMPTZ NOT NULL,
+                    "source" VARCHAR(50) NOT NULL
                 );
             """)
             logger.info("Table 'raw_crypto_data' initialized.")
 
-            // Initialize `transformed_crypto_data` table
             logger.info("Initializing 'transformed_crypto_data' table...")
             jdbcTemplate.execute("""
                 CREATE TABLE IF NOT EXISTS transformed_crypto_data (
                     id SERIAL PRIMARY KEY,
                     symbol VARCHAR(10) NOT NULL,
-                    average_price DECIMAL NOT NULL,
-                    timestamp TIMESTAMPTZ NOT NULL
+                    currency VARCHAR(10) NOT NULL,
+                    open DECIMAL NOT NULL,
+                    high DECIMAL NOT NULL,
+                    low DECIMAL NOT NULL,
+                    close DECIMAL NOT NULL,
+                    volume_from DECIMAL NOT NULL,
+                    volume_to DECIMAL NOT NULL,
+                    timestamp TIMESTAMPTZ NOT NULL,
+                    "source" VARCHAR(50) NOT NULL
```

Comment:
The transformed table should not have the source column. Instead, the transformed table can use data from multiple raw (or source) tables to build its registers. This information is abstract to the final user.